### PR TITLE
MM-19902 - Support for Plugins Menu Items in Channel Header Dropdown

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary

Added support for plugins to register menu items in the Channel Header Dropdown.

#### Ticket Link
[MM-19902](https://mattermost.atlassian.net/browse/MM-19902)

#### Screenshot

Screen Shot 2019-11-28 at 9 19 11 AM

#### Related Pull Requests

- Demo Plugin https://github.com/mattermost/mattermost-plugin-demo/pull/77

### CODE REVIEW

File: components/channel_header_dropdown/channel_header_dropdown_items.js [hunk 1]
```diff
@@ -267,7 +283,9 @@ export default class ChannelHeaderDropdown extends React.PureComponent {
                         />
                     </ChannelPermissionGate>
                 </Menu.Group>
-
+                <Menu.Group>
+                    {pluginItems}
+                </Menu.Group>
                 <Menu.Group divider={divider}>
                     {isMobile &&
                         <MobileChannelHeaderPlug
```
Comment (lieut-data) [inline components/channel_header_dropdown/channel_header_dropdown_items.js]: Unrelated, but I can't remember what this is -- does this overlap with any of your effort?

File: components/channel_header_dropdown/channel_header_dropdown_items.js [hunk 2]
```diff
@@ -267,7 +283,9 @@ export default class ChannelHeaderDropdown extends React.PureComponent {
                         />
                     </ChannelPermissionGate>
                 </Menu.Group>
-
+                <Menu.Group>
```
Comment (lieut-data) [inline components/channel_header_dropdown/channel_header_dropdown_items.js]: Will there be a divider between normal options and plugin options?

Comment (marianunez) [reply components/channel_header_dropdown/channel_header_dropdown_items.js]: Yes, `Menu.Group` includes the divider. See attached screenshot in the PR description.

File: components/channel_header_dropdown/channel_header_dropdown_items.js (hunk 1 above)
Comment (marianunez) [reply components/channel_header_dropdown/channel_header_dropdown_items.js]: I had to look into that myself while I was in there. It seems that the `MobileChannelHeaderPlug` component is used to display the plugins channel header buttons (not to be confused with the channel header menu) in mobile view. 
However, I honestly do not understand why it's being used here too. Maybe @jespino has some background on this.

**Pull request merged**